src/proxy/ClientBuild.php

```diff
@@ -64,29 +74,33 @@ public function __construct(Command $command, array $config = [])
     public function init()
     {
         parent::init();
-        
-        if ($this->_buildConfig === null) {
-            throw new InvalidConfigException("build config can not be empty!");
+    
+        if ($this->db) {
+            $this->db = Instance::ensure($this->db, Connection::class);
+        } else {
+            $this->db = Yii::$app->db;
         }
+        
+        $this->initBuildConfig();
     }
     
-    private $_buildConfig;
-
-    public function setBuildConfig(array $config)
+    public function initBuildConfig()
```

Comment:
@nadar this is needed because in line 100 the db connection is called `$schema = $this->db->getTableSchema($tableName);`
And before that the `$this->db` have to be initialized.

Reply:
Yes but we should do the instantiate with `$db='db'` then the if else is not needed. Also i assume the methods should be a setter method, this ensures that an array has to be provided for buildConfig. So i assume there is no need to rename setBuildConfig just add the Instance::ensure after init().

Reply:
Or you instantiate the object already in the command, then the ensure und creation of the db object is not needed at this point!